### PR TITLE
Add escape hatch to run arbitrary setup script

### LINKED ISSUE #11: There is no facility for installing additional unpackaged non-Python dependencies
If I want to build manylinux2010 wheel for a project that requires the Rust build toolchain then it seems I am somewhat out of luck.  The manylinux2010 build environment has no system packages for the Rust build toolchain and I can't install it myself using the parameters exposed by this Github action.

It would be useful to have an escape hatch for executing an arbitrary executable (maybe a shell script) in the build environment as part of setup.

### PR DESCRIPTION
**Pull request by exarkun**
Fixes #11

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -15,6 +15,10 @@ inputs:
     description: 'yum packages to install, space-separated'
     required: false
     default: ''
+  install-extra-requirements:
```
Comment (RalfG) [inline action.yml]:
```suggestion
  pre-build-script:
```

File: action.yml [hunk 2]
```diff
@@ -31,6 +35,7 @@ runs:
     - ${{ inputs.build-requirements }}
     - ${{ inputs.system-packages }}
     - ${{ inputs.package-path }}
+    - ${{ inputs.install-extra-requirements }}
```
Comment (RalfG) [inline action.yml]:
```suggestion
    - ${{ inputs.pre-build-script }}
```

File: entrypoint.sh [hunk 3]
```diff
@@ -6,12 +6,17 @@ PY_VERSIONS=$1
 BUILD_REQUIREMENTS=$2
 SYSTEM_PACKAGES=$3
 PACKAGE_PATH=$4
-PIP_WHEEL_ARGS=$5
+INSTALL_EXTRA_REQUIREMENTS=$5
```
Comment (RalfG) [inline entrypoint.sh]:
```suggestion
PRE_BUILD_SCRIPT=$5
```

File: entrypoint.sh [hunk 4]
```diff
@@ -6,12 +6,17 @@ PY_VERSIONS=$1
...
+PIP_WHEEL_ARGS=$6
 
 if [ ! -z "$SYSTEM_PACKAGES" ]; then
     yum install -y ${SYSTEM_PACKAGES}  || { echo "Installing yum package(s) failed."; exit 1; }
 fi
 
+if [ ! -z "$INSTALL_EXTRA_REQUIREMENTS" ]; then
```
Comment (RalfG) [inline entrypoint.sh]:
```suggestion
if [ ! -z "$PRE_BUILD_SCRIPT" ]; then
```

File: entrypoint.sh [hunk 5]
```diff
@@ -6,12 +6,17 @@ PY_VERSIONS=$1
...
+    "$INSTALL_EXTRA_REQUIREMENTS"
```
Comment (RalfG) [inline entrypoint.sh]:
```suggestion
    "$PRE_BUILD_SCRIPT" || { echo "Pre-build script failed."; exit 1; }
```

**Review by RalfG — CHANGES_REQUESTED** [review]
Great idea. Thanks for the PR!

I think `pre-build-script` would be a better name for the option, as it requires a full command, not just a list of requirements, such as `build-requirements` and `system-packages`. Can you implement the suggested changes, and additonally update `README.md` and `full_workflow_example.yml` accordingly? Thanks!

**Comment by exarkun** [comment]
> I think pre-build-script would be a better name for the option, as it requires a full command, not just a list of requirements, such as build-requirements and system-packages. Can you implement the suggested changes, and additonally update README.md and full_workflow_example.yml accordingly? Thanks!

Thanks.  I think that is a better name, indeed.

Since I submitted the PR I went ahead and did some more testing of the changes for my project and I found I needed to make one further change - executing the pre-build script was not sufficient because I also needed to get a new directory into PATH for the build step (my use-case is installing the Rust `cargo` tool).

I fixed this by making the pre-build script get _sourced_ instead of executed so it can export environment changes to entrypoint.sh.  I'm not sure how sound this approach is, though.  I wonder if you have any thoughts on this requirement?

**Comment by RalfG** [comment]
Exporting would be yet another option then, next to executing... But I noticed that there is no execution command (i.e. no `sh`) in front of the `"$PRE_BUILD_SCRIPT"` variable:

```
if [ ! -z "$PRE_BUILD_SCRIPT" ]; then
    "$PRE_BUILD_SCRIPT" || { echo "Pre-build script failed."; exit 1; }
fi
```

So instead of positioning this option as the path to a script, this could just be a custom command. In your case `pre-build-script: "source <path_to_script.sh>"` should work, right? For others this could be `sh <path_to_script.sh>`, or even a simple command, e.g. modifying an environment variable: `"CFLAGS=${CFLAGS//-pipe/}"`.

**Review by RalfG — APPROVED** [review]

**Pull request merged**